src/MongoDB.Driver/Linq/Linq3Implementation/Ast/Expressions/AstExpression.cs

```diff
@@ -732,13 +716,18 @@ static bool IsSimple(AstExpression expression)
                 return
                     expression == null ||
                     expression.NodeType == AstNodeType.ConstantExpression ||
-                    expression.NodeType == AstNodeType.FieldExpression;
+                    expression.CanBeRenderedAsFieldPath();
             }
         }
 
-        public static AstVar Var(string name, AstExpression value)
+        public static AstVarExpression Var(string name, bool isCurrent = false)
         {
-            return new AstVar(name, value);
+            return new AstVarExpression(name, isCurrent);
+        }
+
+        public static AstVarBinding VarBinding(string name, AstExpression value)
```

Comment:
`VarBinding` associates a value with a `Var`. There are many places where we have code like this where we define a `Var` with a particular name and then a `VarBinding` with the same name:
```
var stringAst = AstExpression.Var("string");
var stringVar = AstExpression.VarBinding("string", stringTranslation.Ast);
```
Consider whether `name` should be `AstVarExpression` rather than a `string`:
```
public static AstVarBinding VarBinding(AstVarExpression name, AstExpression value)
```